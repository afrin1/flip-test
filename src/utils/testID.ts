import { Platform } from 'react-native'

export default function testID(id: string) {
  return Platform.OS === 'android'
    ? { accessible: true, accessibilityLabel: id }
    : { testID: id }
}
