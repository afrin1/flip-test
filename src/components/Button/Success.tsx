import React from "react";
import { StyleSheet, Text, TouchableOpacity } from "react-native";
import { ButtonProps } from ".";
import { colors } from "../../theme/colors";
import { buttonText } from "../../theme/typography";
import testID from "../../utils/testID";

const SuccessButton = (props: ButtonProps) => {
  const { title, onPress } = props
  return (
    <TouchableOpacity
      {...testID('success-button')}
      onPress={onPress}
      style={styles.button}>
      <Text style={styles.text}>{title}</Text>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  button: {
    justifyContent: 'center',
  },
  text: {
    color: '#FFFFFF',
    backgroundColor: colors.SUCCESS,
    borderRadius: 10,
    padding: 10,
    margin: 5,
    ...buttonText as any
  },
})

export default SuccessButton;