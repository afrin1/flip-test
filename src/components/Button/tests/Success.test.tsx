import { fireEvent, render } from '@testing-library/react-native'
import React from 'react'
import testID from '../../../utils/testID'
import Success from '../Success'

describe('Success Button', () => {
  it('should match snapshot', () => {
    expect(render(
      <Success title={'title'} onPress={jest.fn()} />)
    ).toMatchSnapshot()
  })

  it('should display the provided text', () => {
    const { getByText } = render(<Success title={'title'} onPress={jest.fn()} />)
    expect(getByText('title')).not.toBeNull()
  })

  it('should invoke the provided callback when button is pressed', () => {
    const onPressMock = jest.fn()
    const { getByTestId } = render(
      <Success 
      {...testID('success-button')}
      title={'title'} 
      onPress={onPressMock} />
    )

    fireEvent.press(getByTestId('success-button'))
    expect(onPressMock).toBeCalled()
  })
})