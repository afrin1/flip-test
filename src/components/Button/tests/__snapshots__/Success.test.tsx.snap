// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success Button should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "justifyContent": "center",
    }
  }
  title="title"
>
  <Text
    style={
      Object {
        "backgroundColor": "#00AF33",
        "borderRadius": 10,
        "color": "#FFFFFF",
        "fontSize": 15,
        "fontWeight": "bold",
        "margin": 5,
        "padding": 10,
        "textAlignVertical": "center",
      }
    }
  >
    title
  </Text>
</View>
`;
